### PR TITLE
Fix/BaseSkillTestCase.setup

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

~~changes the `setup` method on `BaseSkillTestCase` from a class method into a regular method. This will only affect mypy signature checking; I have chosen not to change it into `setup_class` since it would affect code execution logic.~~

After investigation it seems logical to turn the method, which is supposed to setup the class, into `setup_class` instead. With respect to backwards incompatibility, I suggest to introduce a setup (non-class method) that raises a `DeprecationWarning` :
```
def setup(self, **kwargs: Any) -> None:
    warn("Use .setup_class or overwrite this method", DeprecationWarning, level=2)
    self.setup_class(**kwargs)
````

references:
- [python docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp)
- [SO ref](https://stackoverflow.com/questions/23667610/what-is-the-difference-between-setup-and-setupclass-in-python-unittest)

someone knows what's on
https://github.com/valory-xyz/open-aea/blob/de7df5614786a0583aec28fa5a53371d09018e3e/aea/test_tools/test_skill.py#L60-L61

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Comment by Karrenbelt** [comment]
Failing tests are those that expect a fresh `setup` and, looking into next.
```
=========================== short test summary info ============================
FAILED tests/test_cli/test_get_address.py::TestGetAddressCommand::test_get_address[fake-password]
FAILED tests/test_cli/test_get_wealth.py::TestGetWealth::test_get_wealth[None]
FAILED tests/test_test_tools/test_test_skill.py::TestSkillTestCase::test_get_quantity_in_outbox
FAILED tests/test_test_tools/test_test_skill.py::TestSkillTestCase::test_drop_messages_from_outbox
FAILED tests/test_test_tools/test_test_skill.py::TestSkillTestCase::test_assert_quantity_in_decision_making_queue
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeMissingKey::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandConnectionPositive::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandConnectionIdPositive::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandConnectionIdURIPositive::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeBadUri::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeFullMultiaddrComputation::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandPositive::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandConnectionNegative::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeBadPortField::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeBadConnectionId::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativePeerId::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeOnlyHostSpecified::test_run[None]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeUriNotExisting::test_run[fake-password]
FAILED tests/test_cli/test_get_multiaddress.py::TestGetMultiAddressCommandNegativeBadHostField::test_run[None]
= 19 failed, 2328 passed, 120 skipped, 59 deselected, 9 warnings, 2 rerun in 1492.29s (0:24:52) =
```

File: aea/test_tools/test_skill.py [hunk 1]
```diff
@@ -458,7 +458,7 @@ def prepare_skill_dialogue(
         return dialogue
 
     @classmethod
-    def setup(cls, **kwargs: Any) -> None:
```
Comment (DavidMinarsch) [inline aea/test_tools/test_skill.py:461]: The base class should still define a `setup` method. Check what can be moved in there..

File: packages/fetchai/skills/http_echo/tests/test_handlers.py [hunk 2]
```diff
@@ -44,9 +44,9 @@ class TestHttpHandler(BaseSkillTestCase):
     path_to_skill = PACKAGE_DIR
 
     @classmethod
-    def setup(cls):
+    def setup_class(cls):
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/http_echo/tests/test_handlers.py:47]: Not all the below seems a class level setup. E.g. the message list `list_of_messages` should stay at `setup` level imo

File: packages/fetchai/skills/generic_seller/tests/test_behaviours.py [hunk 3]
```diff
@@ -47,14 +47,16 @@ class TestSkillBehaviour(BaseSkillTestCase):
...
-    def setup(cls):
+    def setup_class(cls):
         """Setup the test class."""
-        super().setup()
+        super().setup_class()
         cls.service_registration = cast(
             GenericServiceRegistrationBehaviour,
             cls._skill.skill_context.behaviours.service_registration,
         )
+        cls._init_service_registration = cls.service_registration.__dict__.copy()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_seller/tests/test_behaviours.py:57]: Why not do it in a new `setup`?

File: packages/fetchai/skills/generic_buyer/tests/test_models.py [hunk 4]
```diff
@@ -76,6 +76,13 @@ def setup(cls):
             name="strategy",
             skill_context=cls._skill.skill_context,
         )
+        cls._init_strategy_kwargs = cls.strategy.__dict__.copy()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/tests/test_models.py:79]: Why not setup `strategy` in new `setup`?

File: packages/fetchai/skills/generic_buyer/tests/test_handlers.py [hunk 5]
```diff
@@ -569,6 +570,12 @@ def setup(cls):
             ),
         )
 
+    def teardown(self) -> None:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_buyer/tests/test_handlers.py:573]: Already available on base

File: packages/fetchai/skills/erc1155_deploy/tests/intermediate_class.py [hunk 6]
```diff
@@ -153,6 +154,7 @@ def setup(cls):
 
         # models
         cls.strategy = cast(Strategy, cls._skill.skill_context.strategy)
+        cls._init_strategy_kwargs = cls.strategy.__dict__.copy()
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_deploy/tests/intermediate_class.py:157]: do it in new `setup`...

File: packages/fetchai/skills/erc1155_client/tests/test_behaviours.py [hunk 7]
```diff
@@ -52,7 +52,8 @@ def test_setup(self):
     def test_act_is_searching(self):
         """Test the act method of the search behaviour where is_searching is True."""
         # setup
-        self.strategy._is_searching = True
+        self.strategy.is_searching = True
+        assert self.strategy.is_searching
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_client/tests/test_behaviours.py:56]: remove assert

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**